### PR TITLE
[IMP] project : Add project Tree view, improve some wordings and hide uom conversion when default and selected are the same

### PR DESCRIPTION
**Pull request by RaoufGhrissi**
Before this changes:
- In the product form, when the selected uom is Hours and as Hours is the default uom, we display this conversion (1.0 Hours = 1.0 Hours) which is useless
- Kanban is the only view available for projects. With this view, we can't select a batch of projects and display optional fields
- Some wordings didn't have a significant meaning.

This changes hides uom conversion when selected and default uom are the same.
This changes adds the projects tree view.
This changes improves some wordings according to what was mentionned in the task.

task-2724965

### CODE REVIEW

**Review by xavierbol — COMMENTED** [review]
LGTM, thanks for your work.

Can you squash your commits into one and write a commit message? Thanks in advance! 🙂

**Comment by RaoufGhrissi** [comment]
> LGTM, thanks for your work.
> 
> Can you squash your commits into one and write a commit message? Thanks in advance! slightly_smiling_face

**Comment by RaoufGhrissi** [comment]
@xavierbol comments fixed for both community and enterprise

File: addons/project/views/project_views.xml [hunk 1]
```diff
@@ -633,7 +635,7 @@
             <field name="target">new</field>
             <field name="context">{"default_allow_billable": 1}</field>
         </record>
-
+        <!-- here -->
```
Comment (IT-Ideas) [inline addons/project/views/project_views.xml:638]: I guess it is a forgetting 🙂

File: addons/project/models/project.py [hunk 2]
```diff
@@ -686,10 +686,7 @@ def action_view_tasks_analysis(self):
 
     @api.model
     def _action_open_all_projects(self):
-        action = self.env['ir.actions.act_window']._for_xml_id(
-            'project.open_view_project_all' if not self.user_has_groups('project.group_project_stages') else
-            'project.open_view_project_all_group_stage')
-        return action
+        return self.env['ir.actions.act_window']._for_xml_id('project.open_view_project_all')
```
Comment (IT-Ideas) [inline addons/project/models/project.py:689]: I am not sure this is still useful to have this def if it is only to return the action from the xml_id and as it is only used once in the code.
We could use `self.env['ir.actions.act_window']._for_xml_id('project.open_view_project_all')` directly in `confirm_delete` of `project.delete.wizard`

**Review by IT-Ideas — COMMENTED** [review]
Thanks for your work 🙂

File: addons/web/static/src/legacy/scss/form_view.scss [hunk 3]
```diff
@@ -60,6 +60,9 @@ $o-form-label-margin-right: 0px;
             }
         }
     }
+    .zero_min_height { 
+        min-height: 0px !important; 
+    }
```
[the remark this answers is not in the document]
Comment (RaoufGhrissi) [reply addons/web/static/src/legacy/scss/form_view.scss:65]: due to this small flicker 
https://watch.screencastify.com/v/YoGAVORJNCqrobJh29nT

[the remark this answers is not in the document]
Comment (IT-Ideas) [reply addons/web/static/src/legacy/scss/form_view.scss:65]: I did not realized but you actually made a change in a file in `web` module. Those files are intender to be generic. When really needing a custo, prefer updating css within modules as those will only be present once the app is there.
I will push an extra commit in order to show you an alternative that does not require css and is more standard.

**Review by IT-Ideas — COMMENTED** [review]
Thanks. I will push a small fixup commit to show you an alternative to the css trick. Once you have seen it, you can squash it into yours 🙂 .

**Comment by IT-Ideas** [comment]
@robodoo override=ci/security

Reason markup is used on a static text that is safe. Only the text content was updated in this PR.

**Comment by IT-Ideas** [comment]
@robodoo r+

**Pull request closed without merging**